### PR TITLE
Next gen init

### PR DESCRIPTION
**Pull request by Jpunt**
This introduces the initialisation of a server/client-project, in one of the following ways:

```
$ npx samen init client
$ npx samen init server
$ npx samen init // choose client or server
$ npx samen // same as `npx samen init` when no samen is found
```

https://user-images.githubusercontent.com/190501/179243065-5ac6b444-440f-4467-863d-aa6d15b4a940.mov

### CODE REVIEW

**Review by kamilafsar — APPROVED** [review]
Very nice addition man! This will certainly help users get started!
One thing I'm not entirely sure about is that `npx samen` defaults to `npx samen init` if there's no project found. 
Initialisation of projects is a relatively rare need. When you want that, it's okay to add an `init` to the command. When you run `npx samen` you probably just want to run your existing project. If there's no project to be found, we could just say "no samen projects detected, run samen init to create one". 

I'm not *against* defaulting to `init` but I'm not entirely sure about the consequences. Maybe KISS for now... dunno.. What do you think?

**Comment by Jpunt** [comment]
> Very nice addition man! This will certainly help users get started!
> One thing I'm not entirely sure about is that npx samen defaults to npx samen init if there's no project found.
> Initialisation of projects is a relatively rare need. When you want that, it's okay to add an init to the command. When you run npx samen you probably just want to run your existing project. If there's no project to be found, we could just say "no samen projects detected, run samen init to create one".
> 
> I'm not against defaulting to init but I'm not entirely sure about the consequences. Maybe KISS for now... dunno.. What do you think?

Actually, that's a lot better! I'll go fix 👍

File: packages/cli/src/components/Init.tsx [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+
+async function createArticle(article: Article): Promise<void> {
+  fakeDb = [...fakeDb, article]
+}
+
+async function deleteArticle(id: string): Promise<void> {
+  fakeDb = fakeDb.filter((article) => article.id !== id)
+}
+
+export const articleService = createService({
```
[the remark this answers is not in the document]
Comment (Jpunt) [reply packages/cli/src/components/Init.tsx:48]: Agreed!

File: packages/cli/src/components/Init.tsx [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+                  Samen is not installed in this directory. What do you want to
+                  do?
+                </Text>
+              </Box>
+
+              <SelectInput
+                items={[
+                  { value: OptionValue.InitServer, label: "Initialize server" },
+                  { value: OptionValue.InitClient, label: "Initialize client" },
+                  { value: OptionValue.ShutDown, label: "Do nothing" },
```
[the remark this answers is not in the document]
Comment (Jpunt) [reply packages/cli/src/components/Init.tsx]: Agreed, but with a capital E 😉

File: packages/cli/src/components/DevEnv.tsx [hunk 3]
```diff
@@ -51,13 +52,24 @@ export default class DevEnv extends React.Component<Props, State> {
...
+  const updateProjects = useCallback(async () => {
+    try {
+      setProjects(await getProjects())
+    } catch (error) {
+      fatalError(error)
+    }
+  }, [])
+
+  useEffect(() => {
+    updateProjects()
```
[the remark this answers is not in the document]
Comment (Jpunt) [reply packages/cli/src/components/DevEnv.tsx]: Irrelevant due to https://github.com/samen-io/samen/pull/23/commits/5682c08a07263f3b72b22d7d08800245577e2700

**Comment by Jpunt** [comment]
@kamilafsar great points, fixed!

**Comment by kamilafsar** [comment]
Nice! Merged 💪

**Pull request merged**